src/eth_provider/error.rs

```diff
@@ -219,6 +219,9 @@ pub enum TransactionError {
     /// Thrown when the chain id is invalid.
     #[error("invalid chain id")]
     InvalidChainId,
+    /// Thrown when the transaction type is invalid.
+    #[error("invalid transaction type")]
+    InvalidTransactionType,
```

Comment:
As a followup good first issue, maybe we can improve some of these errors. For example:
- `InvalidTransactionType` could come with an error message indicating precisely the transaction type in question.
- I think it's the same for `InvalidChainId` where we could put an error message saying: "expected chain ID: ...., obtained chain ID: ...`

in order to give more information to these errors. If you agree I can open an issue about this.

Reply:
Agreed!

Reply:
Done https://github.com/kkrt-labs/kakarot-rpc/issues/1112